### PR TITLE
Applications list filtering

### PR DESCRIPTION
**Pull request by jaleszek**
Application filters.
Separation the filtering list for admin and assessor.
Additional filtering options.

### CODE REVIEW

File: app/controllers/concerns/form_answer_mixin.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
+module FormAnswerMixin
+  def update
```
Comment (matthewford) [inline app/controllers/concerns/form_answer_mixin.rb]: Assignment Branch Condition size for update is too high. [18/15]

File: app/search/form_answer_status/filtering_helper.rb [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+      [v[:label], k]
+    end
+  end
+
+  def supported_filter_attrs
+    options.keys.map(&:to_s)
+  end
+
+  def all
+    collection.map{|s| s.last.to_s} + sub_collection.map{|s| s.last.to_s}
```
Comment (matthewford) [inline app/search/form_answer_status/filtering_helper.rb]: Space missing to the left of {.<br>Space between { and | missing.<br>Space missing inside }.

File: spec/models/assessor_assignment_spec.rb [hunk 3]
```diff
@@ -151,6 +151,40 @@
...
   end
+
+  context "assessor change" do
+    let(:form) { create(:form_answer, :trade) }
+    context "for assigning the primary/secondary assessor per application" do
+      it "changes the state of flag on application" do
+        primary = form.assessor_assignments.primary
+        secondary = form.assessor_assignments.secondary
+
+        expect {
```
Comment (matthewford) [inline spec/models/assessor_assignment_spec.rb]: Avoid using {...} for multi-line blocks.

File: spec/models/assessor_assignment_spec.rb [hunk 4]
```diff
@@ -151,6 +151,40 @@
...
+          primary.update_attributes(assessor_id: 1)
+        }.to change {
+          form.reload.primary_assessor_not_assigned
+        }.from(true).to(false)
+
+        expect {
```
Comment (matthewford) [inline spec/models/assessor_assignment_spec.rb]: Avoid using {...} for multi-line blocks.

File: spec/models/assessor_assignment_spec.rb [hunk 5]
```diff
@@ -151,6 +151,40 @@
...
+          secondary.update_attributes(assessor_id: 2)
+        }.to change {
+          form.reload.secondary_assessor_not_assigned
+        }.from(true).to(false)
+
+        expect {
```
Comment (matthewford) [inline spec/models/assessor_assignment_spec.rb]: Avoid using {...} for multi-line blocks.

File: spec/models/assessor_assignment_spec.rb [hunk 6]
```diff
@@ -151,6 +151,40 @@
...
+          secondary.update_attributes(assessor: nil)
+        }.to change {
+          form.reload.secondary_assessor_not_assigned
+        }.from(false).to(true)
+
+        expect {
```
Comment (matthewford) [inline spec/models/assessor_assignment_spec.rb]: Avoid using {...} for multi-line blocks.

Comment (jaleszek) [inline spec/models/assessor_assignment_spec.rb]: this method doesn't work with `do-end` blocks :P

File: app/decorators/form_answer_decorator.rb [hunk 7]
```diff
@@ -52,6 +52,7 @@ def award_application_title
   end
 
   def company_or_nominee_name
+    return award_type if Rails.env.development?
```
Comment (jaleszek) [inline app/decorators/form_answer_decorator.rb]: not ideal but really helps with dev

File: spec/models/assessor_assignment_spec.rb (hunk 3 above)
Comment (matthewford) [inline spec/models/assessor_assignment_spec.rb]: Avoid using {...} for multi-line blocks.

File: spec/models/assessor_assignment_spec.rb (hunk 4 above)
Comment (matthewford) [inline spec/models/assessor_assignment_spec.rb]: Avoid using {...} for multi-line blocks.

File: spec/models/assessor_assignment_spec.rb (hunk 5 above)
Comment (matthewford) [inline spec/models/assessor_assignment_spec.rb]: Avoid using {...} for multi-line blocks.

**Comment by jaleszek** [comment]
@RuslanHamidullin or @dreamfall would you have chance to take a look ?

**Pull request merged**